### DISCUSSION TITLE
Understanding lower-level implementation of canary upgrades

### DISCUSSION THREAD

**Discussion by wtzhang23** [istio/istio#51835] [Q&A] [answered]
For a canonical (no remote) deployment, is the meat of the logic done in the sidecar injection webhook? E.g. the sidecar injection webhook watches for deployments in namespaces labelled with the same revision, sets the discoveryAddress to the istiod dns name corresponding to the revision, etc.?

**Comment by howardjohn** [comment, marked as answer]
Yep, the key part is injecting which discovery address to connect to. Each revision can inject different things but most configs in istio are dynamic which is driven by which discovery address we connect to
